### PR TITLE
Jcm sage site updates

### PR DESCRIPTION
**Pull request by cmcfadzean**
## Description
This Introduces some of the changes that the design team has put forward for improvements to the documentation site. Most notable:

- Reintroduction of Panels
- Top Bar Color
- Top Bar Navigation
- Sidebar Sub Navigation
- Bypassing Listing Page for individual Elements & Objects

### Screenshots
Screen Shot 2021-03-30 at 10 59 02 AM
Screen Shot 2021-03-30 at 10 59 16 AM

### Estimated impact
These changes should only affect the documentation with no Kajabi Products Impact.

### CODE REVIEW

**Review by joedakroub — APPROVED** [review]
This looks and functions beautifully! What a significant change, wow. ❤️ 

The only suggestion, not a blocker, would be to highlight the top-level nav section based on where you are:

If I'm on http://0.0.0.0:4000/pages/layout/grid_templates, then the `Layouts` section could be highlighted.

File: docs/lib/sage-frontend/stylesheets/docs/_panel.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.docs-panel {
+  @include sage-panel;
+  padding: 80px;
```
Comment (pixelflips) [inline docs/lib/sage-frontend/stylesheets/docs/_panel.scss:3]: Could we use rem() here?

**Review by pixelflips — COMMENTED** [review]
Also not sure if applicable here, but we usually merge into the `develop` branch.

**Review by pixelflips — APPROVED** [review]
Thanks for the updates @cmcfadzean - LGTM 👍🏼 

One other mention for a possible follow-up, if we don't want to handle it in this PR, is the navigation for the mobile/small screen view. The new top menu items and the icon for expanding on smaller screens are going to need some attention. The toggle button in the header is the wrong color and I'm not exactly sure how we want to handle the new nav across the top of the page.

**Comment by pixelflips** [comment]
@cmcfadzean - sorry my man, a couple more ideas for a follow-up I just came up while navigating around.

1. There doesn't seem to be a way to navigate back to the homepage. (maybe we link the logo?)
2. The Figma iframe on the Design Principles page is causing quite a few console errors. (looks like it's on their side though so might not be much we can do.)

File: docs/lib/sage-frontend/stylesheets/docs/_panel.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.docs-panel {
+  @include sage-panel;
+  padding: 5rem;
```
Comment (monkeypox8) [inline docs/lib/sage-frontend/stylesheets/docs/_panel.scss:3]: I think we'll want this guy under a desktop-only media query. It's fairly large on mobile screens, and in those cases the default padding will work.

**Review by monkeypox8 — COMMENTED** [review]
Love the direction this is heading! As @pixelflips mentioned, the mobile nav appears to be broken. It seems to be a result of the parent flex container, so if a `flex-basis` matching the button width is set on `sage-menubutton`, that should restore its sizing.

Bigger question is what to do with the mobile nav in general. A temporary solution could be to set `sage-assistant` to overflow `auto` rather than `visible` at small screen sizes, but it still doesn't seem ideal having to scroll for main navigation.

One other item to note; there doesn't appear to be a way to navigate back to the home/index, since the Sage logo is not linked.

|  mobile view   |  mobile menu open  |
|--------|--------|
|Simulator Screen Shot - iPhone 8 - 2021-03-30 at 18 05 39 6 08 26 PM|Simulator Screen Shot - iPhone 8 - 2021-03-30 at 18 05 45|

**Comment by QuintonJason** [comment]
I'm not seeing the navigation on desktop. Is anyone experiencing this?
Screen Shot 2021-03-31 at 1 38 27 PM

**Comment by pixelflips** [comment]
> I'm not seeing the navigation on desktop on the index page. Is anyone experiencing this?

@QuintonJason the sidebar only shows sub-pages and the homepage has none. Hopefully, that helps! Do you see it on other pages?

**Comment by QuintonJason** [comment]
> > I'm not seeing the navigation on desktop on the index page. Is anyone experiencing this?
> 
> @QuintonJason the sidebar only shows sub-pages and the homepage has none. Hopefully, that helps! Do you see it on other pages?

Yep, just checking that it was intentional

**Review by QuintonJason — APPROVED** [review]
Good stuff. I have some non-blocking items:
- the `<h5>`s in the sidebar can be elevated to `<h2>`s with type styles to achieve sizing
- there are multiple `<h1>`s in the code demos section of the component pages

File: docs/app/views/application/_sidebar.html.erb [hunk 3]
```diff
@@ -1,190 +1,140 @@
...
-    ]
-  } %>
+  <% if current_page_foundations? %>
+    <h2 class="t-sage-heading-6 docs-sidebar-heading">FOUNDATIONS</h2>
+    <%= sage_component SageNav, { 
+      items: [
+        {
+          text: "Design Principles",
+          link: pages_foundation_path(:design_principles),
+              active: current_page?(pages_foundation_path(:design_principles)),
```
Comment (pixelflips) [inline docs/app/views/application/_sidebar.html.erb:9]: nitpick only - formatting here. (alignment of active.)

File: docs/app/views/application/_sidebar.html.erb [hunk 4]
```diff
@@ -1,190 +1,140 @@
...
+        },
+        {
+          text: "Accessibility",
+          link: pages_foundation_path(:accessibility),
+              active: current_page?(pages_foundation_path(:accessibility)),
```
Comment (pixelflips) [inline docs/app/views/application/_sidebar.html.erb:14]: alignment nitpick again.

**Review by pixelflips — APPROVED** [review]
A couple of small nitpicks (non-blocking), but LGTM! 👍🏼

File: docs/app/views/application/_sidebar.html.erb [hunk 5]
```diff
@@ -1,190 +1,140 @@
...
+        }
+      ]
+    } %>
+  <% end %>
+  <% if current_page_elements? %>
+    <h2 class="t-sage-heading-6 docs-sidebar-heading">ELEMENTS</h2>
+    <ul class="sage-nav__list">
+      <% sorted_sage_elements.each do |element| %>
+        <li>
+          <a class="sage-nav__link" href="<%= pages_element_path(element) %>">
```
Comment (monkeypox8) [inline docs/app/views/application/_sidebar.html.erb:115]: One side effect of the new component paths is that the description content has returned in the urls:
component-url

I'm not sure if it's a quick fix or not. Should be fine as a follow-up as long as we're aware of the current state.

File: docs/lib/sage-frontend/stylesheets/docs/_theme.scss [hunk 6]
```diff
@@ -4,15 +4,28 @@
...
   height: 30px;
 }
 
-// Content
-.sage-content {
-  background: #fff;
+.sage-assistant__body {
+  .sage-btn-group {
+
+    @media screen and (min-width: sage-breakpoint(md-min)) {
```
Comment (monkeypox8) [inline docs/lib/sage-frontend/stylesheets/docs/_theme.scss:14]: Thanks for making these changes! The nav fits much better onscreen. The menu button is still getting cropped at small screen sizes, so as a temporary fix/fast follow we may want to have `sage-assistant__actions` hidden on mobile, which allows everything to fit. The sassdocs/storybook/version items aren't as necessary as the main nav:

**Review by monkeypox8 — APPROVED** [review]
👍🏼  LGTM! Couple of items to follow up on as nice-to-haves.

**Pull request merged**